### PR TITLE
[amazonechocontrol] fix XML escape for speaking

### PR DESCRIPTION
**Pull request by Trinitus01**
fixed: xml escape speak (act, tts, text command)

Signed-off-by: Tom Blum <trinitus01@googlemail.com>

### CODE REVIEW

File: bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java [hunk 1]
```diff
@@ -1326,20 +1327,22 @@ public void playAmazonMusicPlayList(Device device, @Nullable String playListId)
...
             @Nullable Integer ttsVolume, @Nullable Integer standardVolume) {
-        String plainSpeak = speak.replaceAll("<.+?>", " ").replaceAll("\\s+", " ").trim();
-        String plainBody = bodyText.replaceAll("<.+?>", " ").replaceAll("\\s+", " ").trim();
-
-        if (plainSpeak.isEmpty() && plainBody.isEmpty()) {
-            // if there is neither a bodytext nor (except tags) a speaktext, we have nothing to announce
+        speak = speak.replaceAll("\\s+", " ").trim();
+        bodyText = bodyText.replaceAll("\\s+", " ").trim();
+        String plainSpeak = speak.replaceAll("<.+?>", " ").trim();
+        String plainBodyText = bodyText.replaceAll("<.+?>", " ").trim();
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java]:
```suggestion
        String trimmedSpeak = speak.replaceAll("\\s+", " ").trim(); 
        String plainSpeak = trimmedSpeak.replaceAll("<.+?>", " ").trim();
        String plainBodyText = bodyText.replaceAll("\\s+", " ").replaceAll("<.+?>", " ").trim();
```

Codstyle: Do not assign new values to parameters, consider them `final` (event if not explicitly declared). Use a local variable instead.

File: bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java [hunk 2]
```diff
@@ -1326,20 +1327,22 @@ public void playAmazonMusicPlayList(Device device, @Nullable String playListId)
...
+        if (plainSpeak.isEmpty() && plainBodyText.isEmpty()) {
             return;
         }
+        String escapedSpeak = speak.replace(plainSpeak, XmlEscape.escapeXml10(plainSpeak));
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java]:
```suggestion
        String escapedSpeak = trimmedSpeak.replace(plainSpeak, XmlEscape.escapeXml10(plainSpeak));
```

File: bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java [hunk 3]
```diff
@@ -1391,16 +1394,19 @@ private void sendAnnouncement() {
 
     public void textToSpeech(Device device, String text, @Nullable Integer ttsVolume,
             @Nullable Integer standardVolume) {
-        if (text.replaceAll("<.+?>", "").replaceAll("\\s+", " ").trim().isEmpty()) {
+        text = text.replaceAll("\\s+", " ").trim();
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java]: See above

File: bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java [hunk 4]
```diff
@@ -1441,16 +1447,19 @@ private void sendTextToSpeech() {
     }
 
     public void textCommand(Device device, String text, @Nullable Integer ttsVolume, @Nullable Integer standardVolume) {
-        if (text.replaceAll("<.+?>", "").replaceAll("\\s+", " ").trim().isEmpty()) {
+        text = text.replaceAll("\\s+", " ").trim();
```
Comment (J-N-K) [inline bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java]: See above

**Review by J-N-K — CHANGES_REQUESTED** [review]

File: bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java (hunk 1 above)
Comment (Trinitus01) [reply bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java]: done

File: bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java (hunk 2 above)
Comment (Trinitus01) [reply bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java]: done

File: bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java (hunk 3 above)
Comment (Trinitus01) [reply bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java]: done

File: bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java (hunk 4 above)
Comment (Trinitus01) [reply bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/Connection.java]: done

**Review by J-N-K — APPROVED** [review]

**Pull request merged**